galley/pkg/runtime/processor.go

```diff
@@ -200,19 +201,24 @@ loop:
 
 	p.stateStrategy.Reset()
 	close(p.stopped)
-	scope.Debugf("Process.process: Exiting process loop")
+
+	if scope.DebugEnabled() {
```

Comment:
Nit: You shouldn't need this. scope.Debug (not Debugf) should be pretty optimal. Are you seeing a perf problem with this call?

Reply:
Yes ... It appears to be allocating an array each time it's called